### PR TITLE
Use `proxyApp` instead of abciclient.Client in node clients

### PR DESCRIPTION
**Pull request by Ferret-san**
Closes: #744

### CODE REVIEW

**Comment by Ferret-san** [comment]
this also implies an update to the cosmos-sdk fork, which I will take care of and link here

File: node/full.go [hunk 1]
```diff
@@ -423,3 +428,12 @@ func createAndStartIndexerService(
 
 	return indexerService, txIndexer, blockIndexer, nil
 }
+
+func createAndStartProxyAppConns(clientCreator proxy.ClientCreator, logger log.Logger) (proxy.AppConns, error) {
+	proxyApp := proxy.NewAppConns(clientCreator)
+	proxyApp.SetLogger(logger.With("module", "proxy"))
+	if err := proxyApp.Start(); err != nil {
```
[the remark this answers is not in the document]
Comment (Ferret-san) [reply node/full.go:435]: you are right, it should be moved to `OnStart()`

[the remark this answers is not in the document]
Comment (Ferret-san) [reply node/full.go:435]: Apparently not calling `Start()` on the proxyApp leads to nil pointer dereference errors. So moved the call to `Start()` inside of `newFullNode` and `newLightNode` respectively.

File: node/header_exchange.go [hunk 2]
```diff
@@ -9,8 +9,7 @@ import (
 	"github.com/celestiaorg/go-header"
 	goheaderp2p "github.com/celestiaorg/go-header/p2p"
 	goheaderstore "github.com/celestiaorg/go-header/store"
-	"github.com/celestiaorg/go-header/sync"
-	goheadersync "github.com/celestiaorg/go-header/sync"
+	sync "github.com/celestiaorg/go-header/sync"
```
Comment (S1nus) [inline node/header_exchange.go:12]: good catch

**Review by S1nus — APPROVED** [review]

**Comment by gupadhyaya** [comment]
@Ferret-san can you also make a PR to https://github.com/rollkit/cosmos-sdk/ to comply with this PR?

File: node/light.go [hunk 3]
```diff
@@ -88,6 +95,10 @@ func openDatastore(conf config.NodeConfig, logger log.Logger) (ds.TxnDatastore,
 }
 
 func (ln *LightNode) OnStart() error {
+	if err := ln.proxyApp.Start(); err != nil {
```
Comment (gupadhyaya) [inline node/light.go:98]: you need start in both places? `OnStart` and `newLightNode`?

Comment (Ferret-san) [reply node/light.go:98]: good catch, no need to start it in both places

**Comment by Ferret-san** [comment]
> @Ferret-san can you also make a PR to https://github.com/rollkit/cosmos-sdk/ to comply with this PR?

on it

**Comment by gupadhyaya** [comment]
@Ferret-san can you resolve the conflicts? we can review and merge this after.

**Review by Manav-Aggarwal — DISMISSED** [review]
LGTM

File: types/pb/tendermint/abci/types.pb.go
Comment (gupadhyaya) [inline types/pb/tendermint/abci/types.pb.go:1]: no need to push this file. can you remove?

**Review by nashqueue — APPROVED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**